helper/restore_helper.go

```diff
@@ -43,7 +43,16 @@ var (
  * SUBSET type applies when restoring using plugin(if compatible) from uncompressed data with filters
  * NONSEEKABLE type applies for every other restore scenario
  */
-type RestoreReader struct {
+type RestoreReader interface {
+	waitForPlugin() error
+	positionReader(pos uint64, oid int) error
+	copyData(num int64) (int64, error)
+	copyAllData() (int64, error)
+	getFileHandle() *os.File
+	getReaderType() ReaderType
+}
+
+type RestoreReaderImpl struct {
```

Comment:
How about not renaming `RestoreReader` to `RestoreReaderImpl` and use name `IRestoreReader` as interface name to reduce diff?